android/app/src/googleplay/java/org/fossasia/openevent/fragments/MapsFragment.java

```diff
@@ -180,42 +177,65 @@ private void showLocationsOnMap() {
         });
     }
 
-    private void focucOnMarker(Marker marker){
-        marker.showInfoWindow();
-        marker.setIcon(vectorToBitmap(getContext(), R.drawable.map_marker, R.color.color_primary));
-        mMap.animateCamera(CameraUpdateFactory.newLatLngZoom(marker.getPosition(), (float) Math.floor(mMap.getCameraPosition().zoom + 8)));
+    private void focusOnMarker(String locationName) {
+        mMap.animateCamera(CameraUpdateFactory.newLatLngZoom(stringMicrolocationClusterWrapperMap.get(locationName).getPosition(), mMap.getMaxZoomLevel()), new GoogleMap.CancelableCallback() {
+            @Override
+            public void onFinish() {
+                Handler handler = new Handler();
+                handler.postDelayed(() -> {
+                    Marker marker = stringMicrolocationClusterWrapperMap.get(locationName).getMarker();
+                    if (marker != null) {
+                        locationMarker.setIcon(ImageUtils.vectorToBitmap(getContext(), R.drawable.map_marker, R.color.dark_grey));
+                        locationMarker = marker;
+                        marker.showInfoWindow();
+                        marker.setIcon(ImageUtils.vectorToBitmap(getContext(), R.drawable.map_marker, R.color.color_primary));
+                    }
+                }, 500);
```

Comment:
The marker reference is only possible after it is rendered which happens only after the zoom. Hence postDelayed was necessary to show the info window after the rendering.